### ISSUE TITLE
Cannot compile documentation on Ubuntu 15.04

### ISSUE THREAD

**Issue by rHermes** [marcomaggi/vicare#78]
I'm trying to compile the project from the git-repo using these steps:

```
rhermes@rhermes-VirtualBox:~/build$ git clone https://github.com/marcomaggi/vicare
rhermes@rhermes-VirtualBox:~/build$ cd vicare/
rhermes@rhermes-VirtualBox:~/build/vicare$ sh autogen.sh
rhermes@rhermes-VirtualBox:~/build/vicare$ mkdir build
rhermes@rhermes-VirtualBox:~/build/vicare$ cd build/
rhermes@rhermes-VirtualBox:~/build/vicare/build$ ../configure --enable-maintainer-mode --with-pthread
rhermes@rhermes-VirtualBox:~/build/vicare/build$ make
```

The last command gives this ouput:

```
echo '#define BOOTFILE "/usr/local/lib/vicare-scheme/vicare.boot"' >bootfileloc.h
echo '#define EXEFILE "/usr/local/bin/vicare"' >>bootfileloc.h
depbase=`echo lib/features.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..  -I ../src -I ./src  -Wall -g -O2 -MT lib/features.o -MD -MP -MF $depbase.Tpo -c -o lib/features.o ../lib/features.c &&\
mv -f $depbase.Tpo $depbase.Po
/bin/bash ./libtool  --tag=CC   --mode=link gcc -std=gnu99 -Wall -g -O2  -rdynamic  -o features lib/features.o  -lgmp -lrt -lm -ldl  -lpthread -lreadline
libtool: link: gcc -std=gnu99 -Wall -g -O2 -rdynamic -o features lib/features.o  -lgmp -lrt -lm -ldl -lpthread -lreadline
/bin/mkdir -p lib/vicare/platform && ./features >lib/vicare/platform/features.vicare.sls
make  all-am
make[1]: Entering directory '/home/rhermes/build/vicare/build'
make[2]: Entering directory '/home/rhermes/build/vicare/build'
make[2]: Leaving directory '/home/rhermes/build/vicare/build'
Updating ../doc/version-scheme.texi
restore=: && backupdir=".am$$" && \
am__cwd=`pwd` && CDPATH="${ZSH_VERSION+.}:" && cd .. && \
rm -rf $backupdir && mkdir $backupdir && \
if (/bin/bash /home/rhermes/build/vicare/meta/autotools/missing makeinfo --version) >/dev/null 2>&1; then \
  for f in ../doc/vicare-scheme.info ../doc/vicare-scheme.info-[0-9] ../doc/vicare-scheme.info-[0-9][0-9] ../doc/vicare-scheme.i[0-9] ../doc/vicare-scheme.i[0-9][0-9]; do \
    if test -f $f; then mv $f $backupdir; restore=mv; else :; fi; \
  done; \
else :; fi && \
cd "$am__cwd"; \
if /bin/bash /home/rhermes/build/vicare/meta/autotools/missing makeinfo   -I doc -I ../doc \
 -o ../doc/vicare-scheme.info ../doc/vicare-scheme.texi; \
then \
  rc=0; \
  CDPATH="${ZSH_VERSION+.}:" && cd ..; \
else \
  rc=$?; \
  CDPATH="${ZSH_VERSION+.}:" && cd .. && \
  $restore $backupdir/* `echo "./../doc/vicare-scheme.info" | sed 's|[^/]*$||'`; \
fi; \
rm -rf $backupdir; exit $rc
../doc/expander.texi:3131: macro `stx' declared without argument called with an argument
Makefile:6726: recipe for target '../doc/vicare-scheme.info' failed
make[1]: *** [../doc/vicare-scheme.info] Error 1
make[1]: Leaving directory '/home/rhermes/build/vicare/build'
Makefile:6320: recipe for target 'all' failed
make: *** [all] Error 2
```
I tried googleing for macro "stx", but couldn't find anythong. Note that it is only the documentation part of the build that fails, if I run the process with make -j4 the documentation will still fail, but the vicare binary and co will compile just fine.

I hope I have provided enough information, please say if you need any more information on the system.

**Comment by marcomaggi** [comment]
Thanks for the report.  This should be fixed in the current head of the ```master``` branch.

The problem is that I defined a Texinfo macro called ```stx``` in the file ```doc/macros.texi```; the macro takes zero arguments, but I called it with one argument in the file ```doc/expander.texi```.  This is my true fault: for speed of building, I am still using an old version of Texinfo, which does less input validations; the lastest releases of Texinfo catch this error.
